src/nl/surf/eduhub_rio_mapper/rio/loader.clj

```diff
@@ -60,8 +60,9 @@
   (let [code (when (goedgekeurd? element)
                   ;; TODO: this is ugly, but we don't know at this stage what entity we tried to resolve.
                   (or (single-xml-unwrapper element "ns2:opleidingseenheidcode")
-                      (single-xml-unwrapper element "ns2:aangebodenOpleidingCode")))]
-    (log-rio-action-response (str "RESOLVE:" code) element)
+                      (single-xml-unwrapper element "ns2:aangebodenOpleidingCode")))
+        id   (or code (-> element xml-utils/element->edn :opvragen_rioIdentificatiecode_response :foutmelding :sleutelgegeven :sleutelwaarde))]
+    (log-rio-action-response (str "RESOLVE:" id) element)
```

Comment:
I don't see how this is better.  Please replace `when` on line 60 and log either an error or some response instead of `code` or `id` and some generic log statement.